README.md

```diff
@@ -1,3 +1,132 @@
 # ZMK Studio
 
-Initial work on the ZMK Studio UI.
+ZMK Studio is a powerful and user-friendly graphical interface for configuring and managing ZMK firmware for custom keyboards. It provides a seamless experience for users to customize their keyboard layouts, manage key bindings, and interact with their devices.
+
+## Overview
+
+ZMK Studio offers a range of capabilities to enhance the user experience:
+
+- **Graphical Keymap Editor**: Easily create and modify keymaps with a visual editor.
+- **Device Management**: Connect to your keyboard via USB or Bluetooth and manage its settings.
+- **Firmware Updates**: Flash new firmware to your device directly from the application.
+- **Real-time Feedback**: Receive real-time notifications and feedback from your device.
+
+For more detailed information on the features and capabilities of ZMK Studio, please visit the [ZMK Studio Documentation](https://zmk.dev/docs/features/studio#capabilities).
+
+## Getting Started
+
+To contribute to ZMK Studio, follow these steps:
+
+### Prerequisites
+
+Ensure you have the following installed on your system:
+
+- [Node.js](https://nodejs.org/) (version 14 or later)
+- [npm](https://www.npmjs.com/) (version 6 or later)
+- [Rust](https://www.rust-lang.org/tools/install) (for building Tauri applications)
+- [Tauri CLI](https://tauri.app/v1/guides/getting-started/prerequisites/)
+
+### Setup
+
+1. **Clone the repository**:
+
+   ```sh
+   git clone https://github.com/zmkfirmware/zmk-studio.git
+   cd zmk-studio
+   ```
+
+2. **Install dependencies**:
+
+   ```sh
+   npm install
+   ```
+
+3. **Run the development server**:
+
+   ```sh
+   npm run dev
+   ```
+
+   This will start the Vite development server and open the application in your default web browser.
+
+### Building the Project
+
+To build the project for production, run:
+
+```sh
+npm run build
+```
+
+This will compile the TypeScript code and bundle the application using Vite.
+
+### Running Storybook
+
+To develop and test UI components in isolation, you can use Storybook:
+
+1. **Start Storybook**:
+
+   ```sh
+   npm run storybook
+   ```
+
+   This will start the Storybook server and open it in your default web browser.
+
+2. **Build Storybook**:
+
+   ```sh
+   npm run build-storybook
+   ```
+
+   This will build the Storybook static site for deployment.
+
+### Linting
+
+To ensure code quality and consistency, you can run the linter:
+
+```sh
+npm run lint
+```
+
+This will run ESLint on the codebase and report any issues.
+
+### Tauri Commands
+
+For Tauri-specific commands, you can use:
+
+1. **Run Tauri development server**:
+
+   ```sh
+   npm run tauri dev
```

Comment:
**issue:** this will not work, it should be `npm run tauri -- dev` 😃

Reply:
are you sure it won't work?  It seems to work for me on the `main` branch.

<img width="1510" alt="image" src="https://github.com/user-attachments/assets/5997effe-8620-4d68-bebc-feb83bab690f" />

Reply:
but I'm new to Tauri so if `-- dev` is the preferred approach, I'm happy to update that as well.  So `npm run tauri dev` doesn't work on your machine when you try it?

I'm on MacOS btw. Curious which OS you're running if it doesn't work for you?

Reply:
Actually you're right, it's not a `tauri`  convention but an `npm` one for passing arguments to the command.  I don't understand why my command was working (even `npm run tauri build` works) but what you're suggesting does seem like the more correct syntax.

Updated in https://github.com/zmkfirmware/zmk-studio/pull/114/commits/1ff1c513b2bfc20e6dff1cdebaa785fabc76d8ae

Reply:
i'm on mac as well, tried again and it works. i wonder why it didn't in the beginning. you can ignore my comment.

Reply:
as a counterpoint, [tauri docs omit](https://v2.tauri.app/develop/#developing-your-desktop-application) the `--`. I don't think it matters too much though